kong/cache.lua

```diff
@@ -0,0 +1,164 @@
+local kong_mlcache = require "kong.mlcache"
+
+
+local type    = type
+local max     = math.max
+local ngx_log = ngx.log
+local ngx_now = ngx.now
+local ERR     = ngx.ERR
+local NOTICE  = ngx.NOTICE
+local DEBUG   = ngx.DEBUG
+
+
+local SHM_CACHE = "cache"
```

Comment:
yes, "namespace" is colloquial, referring to usage within the shm. it may be possible that other users want to use this shm? we should be a good citizen imo and be clear about what our keys are doing in the shm.